pymc_marketing/mmm/components/adstock.py

```diff
@@ -81,13 +82,20 @@ class AdstockTransformation(Transformation):
     prefix: str = "adstock"
     lookup_name: str
 
+    @validate_call
     def __init__(
         self,
-        l_max: int,
-        normalize: bool = True,
-        mode: ConvMode = ConvMode.After,
-        priors: dict | None = None,
-        prefix: str | None = None,
+        l_max: int = Field(
+            ..., gt=0, description="Maximum lag for the adstock transformation."
+        ),
+        normalize: bool = Field(
+            True, description="Whether to normalize the adstock values."
+        ),
+        mode: ConvMode = Field(ConvMode.After, description="Convolution mode."),
+        priors: dict | None = Field(
```

Comment:
We can tighten up this definition to dict[str, Prior] and deprecate the conversion. Maybe the conversion function we have can be used in the field validator

Reply:
or would dict[str, InstanceOf[Prior]] work?

Reply:
done!